### DISCUSSION TITLE
Tox on development code

### DISCUSSION THREAD

**Discussion by raysalemi** [tox-dev/tox#2269] [Q&A]
Hi,

When I run tox in my development directory, it insists on installing the version of my package from pypi.org.  The PyPI version is, of course, the old code, and it doesn't match the current tests.

How do I get tox to run with the code under development?

Ray

**Comment by gaborbernat** [comment]
Can you post your configuration and output so we can understand what you're trying to do and what's happening?

**Reply by raysalemi** [reply]
[toxini.txt](https://github.com/tox-dev/tox/files/7491217/toxini.txt)

Sorry, I misspoke.  My situation is that I'm developing code that depends upon a development version of another package that is on my computer.  But when I run tox, it insists on downloading the released version of the package.  So my tests fail because they are running with the released version instead of the version under development.

Is there a way to get tox to run with the packages as my development environment sees them?

**Reply by gaborbernat** [reply]
> But when I run tox, it insists on downloading the released version of the package.

This because you specify that dependency as install requires dependency. If you want to make it development dependency you'd need to install that package in dev mode via `-e path/to/local/source` (while removing the install requires format). tox is only doing what you asked it to do.

**Comment by raysalemi** [comment]
So you can set the install command?  I'll find where.
